src/tools/genpolicy/rules.rego

```diff
@@ -105,7 +105,13 @@ CreateContainerRequest:= {"ops": ops, "allowed": true} {
     ret := allow_linux(ops_builder2, p_oci, i_oci)
     ret.allowed
 
-    ops := ret.ops
+    # save to policy state
+    # key: input.container_id
+    # val: index of p_container in the policy_data.containers array
+    print("CreateContainerRequest: addding container_id=", input.container_id, " to state")
+    add_p_container_to_state := state_allows(input.container_id, idx)
```

Comment:
I just realized that this might not cover the attack surface completely:

1. CreateContainer request with `container_id = deadbeef, policy container 1` - success.
2. CreateContainer request with `container_id = deadbeef, policy container 2` - passes policy, but fails at the agent due to `deadbeef` existing.
3. ExecProcess `container_id = deadbeef, exec from policy container 2` - success(?).

However, since we're not making matters worse, we can leave that as a follow-up improvement.

Reply:
Now I'm confused: how does this actually work? `state_allows` returns a null op if the key-value pair is already in state, and fails otherwise. But where do we actually set the key-value pair?

Reply:
[state_allows_1](https://github.com/kata-containers/kata-containers/blob/c75ea2582e373036681d28b55e6e9daccdd378a6/src/tools/genpolicy/rules.rego#L156) allows for adding state (op = Add)

[state_allows_2](https://github.com/kata-containers/kata-containers/blob/c75ea2582e373036681d28b55e6e9daccdd378a6/src/tools/genpolicy/rules.rego#L169) validates that the an existing key's value is not changed.

Reply:
Thanks both for the explanation, I was misled by the `state_allows` name not sounding mutable.